pkg/registration/registration_client.go

```diff
@@ -228,11 +228,47 @@ func (rClient *K8sRegistrationClient) GetActionMergePolicy() []*proto.ActionMerg
 		ForCommodity(proto.CommodityDTO_VMEM).
 		ForCommodity(proto.CommodityDTO_VCPU_REQUEST).
 		ForCommodity(proto.CommodityDTO_VMEM_REQUEST).
-		DeDuplicateAndAggregateBy(actionMergeTarget).
-		DeDuplicateAndAggregateBy(actionMergeTarget2)
+		DeDuplicateAndAggregateBy(resizeActionMergeTarget).
+		DeDuplicateAndAggregateBy(resizeActionMergeTarget2)
+
+	//horizontal scale action
+	horizontalScaleActionMergeTarget := builder.NewActionDeDuplicateAndAggregationTargetBuilder().
+		DeDuplicatedBy(builder.NewActionAggregationTargetBuilder(proto.EntityDTO_WORKLOAD_CONTROLLER,
+			proto.ConnectedEntity_AGGREGATED_BY_CONNECTION))
+
+	containerPodDataDaemonSet := proto.EntityDTO_ContainerPodData{
+		ControllerData: &proto.EntityDTO_WorkloadControllerData{
+			ControllerType: &proto.EntityDTO_WorkloadControllerData_DaemonSetData{
+				DaemonSetData: &proto.EntityDTO_DaemonSetData{
+					// Populate the fields specific to DaemonSetData
+				},
+			},
+		},
+	}
+
+	containerPodDataReplicaSet := proto.EntityDTO_ContainerPodData{
+		ControllerData: &proto.EntityDTO_WorkloadControllerData{
+			ControllerType: &proto.EntityDTO_WorkloadControllerData_ReplicaSetData{
+				ReplicaSetData: &proto.EntityDTO_ReplicaSetData{
+					// Populate the fields specific to ReplicaSetData
+				},
+			},
+		},
+	}
+
+	containerHorizontalScaleMerge := builder.NewHorizontalScaleMergeSpecBuilder().
+		ForEntityType(proto.EntityDTO_CONTAINER_POD).
+		ForCommodity(proto.CommodityDTO_VCPU).
+		ForCommodity(proto.CommodityDTO_VMEM).
+		ForCommodity(proto.CommodityDTO_VCPU_REQUEST).
+		ForCommodity(proto.CommodityDTO_VMEM_REQUEST).
```

Comment:
Horizontal scale action are not triggered by resource commodities today. They are triggered by SLO commodities like ResponseTime and Transactions.

Reply:
Yes, thanks! I will make the change.